components/builder-db/src/models/channel.rs

```diff
@@ -159,6 +166,37 @@ impl Channel {
         result
     }
 
+    pub fn list_latest_packages(req: &ListAllChannelPackagesForTarget,
+                                conn: &PgConnection)
+                                -> QueryResult<(String, String, Vec<BuilderPackageIdent>)> {
+        Counter::DBCall.increment();
+        let start_time = Instant::now();
+        let channel = String::from(req.channel.as_str());
+        let target = String::from(req.target);
+
+        let result = origin_packages_with_version_array::table
+            .inner_join(origin_channel_packages::table.inner_join(origin_channels::table))
+            .filter(origin_packages_with_version_array::origin.eq(&req.origin))
+            .filter(origin_channels::name.eq(&channel))
+            .filter(origin_packages_with_version_array::target.eq(&target))
+            .filter(origin_packages_with_version_array::visibility.eq(any(req.visibility)))
+            .select(origin_packages_with_version_array::ident)
```

Comment:
I  think we need a `.distinct_on(origin_packages_with_version_array::name)`.  In my test instance with three packages I got the following result: 

```
> curl http://localhost/v1/depot/channels/smacfarlane/unstable/pkgs/latest?target=x86_64-linux |jq
{
  "channel": "unstable",
  "target": "x86_64-linux",
  "data": [
    {
      "origin": "smacfarlane",
      "name": "foo",
      "version": "0",
      "release": "20200605152530"
    },
    {
      "origin": "smacfarlane",
      "name": "foo",
      "version": "0",
      "release": "20200427172822"
    },
    {
      "origin": "smacfarlane",
      "name": "foo",
      "version": "0",
      "release": "20200427172727"
    }
  ]
}
```